### ISSUE TITLE
Incorrect WordPress.Security.ValidatedSanitizedInput.InputNotValidated in some cases

### ISSUE THREAD

**Issue by slaFFik** [WordPress-Coding-Standards/WordPress-Coding-Standards#1473]
This report is incorrect:
```
phpcs: WordPress.Security.ValidatedSanitizedInput.InputNotValidated: Detected usage of a non-validated input variable: $_POST
```
when highlighting this line of code:

```php
$form_id = (int) $_POST['form_id'];
```

Reason: `(int) $_POST['form_id']` will always be an integer regardless of existence or correct/incorrect value of `$_POST['form_id']`. Casting in this situation is totally fine and is actually a sanitization by itself.

PHP 7.1.0, PHPCS 3.3.1, WPCS 1.0.0

**Comment by dingo-d** [comment]
You're missing the check if this `$_POST['form_id']` exists in the first place

```php
    $form_id = ( isset( $_POST['form_id'] ) ) ? (int) $_POST['form_id'] : '';
    $form_id = ( ! empty( $_POST['form_id'] ) ) ? (int) $_POST['form_id'] : '';
```

In this case you only need to add nonce verification 😉

**Comment by slaFFik** [comment]
No, there is no need in extra call of `isset` when type casting to int like in my example. That's a totally valid code with no errors. If form_id doesn't exist - it will be null and typed to 0.

**Comment by dingo-d** [comment]
In the case that `form_id` doesn't exist you'd get a `Undefined index: form_id` notice. That's why the error is shown. So yes, the error is valid imo.

**Comment by slaFFik** [comment]
Ok, I've rechecked, completely forgot to set `error_reporting(E_ALL);`.

You are correct. Closing this issue.
